src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php

```diff
@@ -34,35 +36,63 @@
 final class RemoteLanguagePackLoader implements LanguagePackLoaderInterface
 {
     /**
-     * The link from which available languages are retrieved.
+     * The languages repository base path
+     *
+     * @TODO : duplicated from {prestashop_repository}/app/config/config.yml;
+     *         used for install context or other cases where no Symfony container is available
      */
-    public const PACK_LINK = 'http://i18n.prestashop-project.org/translations/%ps_version%/available_languages.json';
+    private const LANG_REPOSITORY_BASE_PATH = 'https://i18n.prestashop-project.org';
 
     /**
-     * @var Version
+     * @var Version Prestashop version
      */
     private $version;
 
     /**
-     * @param Version $version
+     * @var string represents the language repository base path
      */
-    public function __construct(Version $version)
+    private $langRepositoryBasePath;
+
+    /**
+     * @param Version $version Prestashop version
+     * @param string $langRepositoryBasePath should become mandatory in future release
+     */
+    public function __construct(Version $version, string $langRepositoryBasePath = self::LANG_REPOSITORY_BASE_PATH)
     {
         $this->version = $version;
+        $this->langRepositoryBasePath = $langRepositoryBasePath;
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return string requested URL
+     */
+    public function getLanguagePackUrl(string $locale = null): string
```

Comment:
The locale should never be null, or else you'd get an interpolation error and a bad URL
```suggestion
    public function getLanguagePackUrl(string $locale): string
```

Reply:
This class has mixed concerns. According to its name, it should serve as a means to download a remote language pack, not provide its URLs. I suggest refactoring the class and its consumers to provide those features.

Reply:
:thinking: You are perfectly right ... I thought I removed this after a comment making me separate this function with `getLanguagePackListUrl` ... I am lost with this PR history, sorry ^^'